File: Difficulty: Easy/Chocolate Distribution Problem/chocolate-distribution-problem.java
import java.util.*;

class Solution {
    public int findMinDiff(ArrayList<Integer> arr, int m) {
        
        int n = arr.size();
        
        // Edge cases
        if (m == 0 || n == 0)
            return 0;
        
        if (m > n)
            return -1;
        
        // Sort the array
        Collections.sort(arr);
        
        int minDiff = Integer.MAX_VALUE;
        
        // Find the minimum difference
        for (int i = 0; i + m - 1 < n; i++) {
            int diff = arr.get(i + m - 1) - arr.get(i);
            minDiff = Math.min(minDiff, diff);
        }
        
        return minDiff;
    }
}

File: Difficulty: Easy/Chocolate Distribution Problem/chocolate-distribution-problem.cpp
#include <algorithm>
#include <vector>
using namespace std;

int minDiff(vector<int> &arr, int k) {
    
    int n = arr.size();
    
    if (k == 0 || n == 0)
        return 0;
        
    if (n < k)
        return -1;
        
    sort(arr.begin(), arr.end());
    
    int ans = INT_MAX;
    
    for (int i = 0; i <= n - k; i++) {
        int diff = arr[i + k - 1] - arr[i];
        ans = min(ans, diff);
    }
    
    return ans;
}
